parameter/SelectionCriterionRule.cpp

```diff
@@ -28,26 +28,19 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 #include "SelectionCriterionRule.h"
-#include "SelectionCriterion.h"
 #include "XmlDomainSerializingContext.h"
 #include "XmlDomainImportContext.h"
-#include "SelectionCriteriaDefinition.h"
-#include "SelectionCriterionTypeInterface.h"
 #include "RuleParser.h"
+#include <criterion/Criterion.h>
+
 #include <assert.h>
 
 #define base CRule
 
 using std::string;
 
-const CSelectionCriterionRule::SMatchingRuleDescription CSelectionCriterionRule::_astMatchesWhen[CSelectionCriterionRule::ENbMatchesWhen] = {
-    { "Is", true },
-    { "IsNot", true },
-    { "Includes", false },
-    { "Excludes", false }
-};
-
-CSelectionCriterionRule::CSelectionCriterionRule() : _pSelectionCriterion(NULL), _eMatchesWhen(CSelectionCriterionRule::EIs), _iMatchValue(0)
+CSelectionCriterionRule::CSelectionCriterionRule() :
+    _pSelectionCriterion(NULL), mMatchesWhenVerb(""), mMatchState{0}
```

Comment:
It wasn't clear that the state was a singleton (and for me it's too bad).
Anyway, I think it's much clearer to keep the State type everywhere.